libraries/Microsoft.Bot.Builder.Core.State/IStateStorageProvider.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.Core.State
+{
+    public interface IStateStorageProvider
+    {
+        IStateStorageEntry CreateNewEntry(string stateNamespace, string key);
```

Comment:
>key [](start = 72, length = 3)

It's not clear if the key needs to be unique at time of creation. I'm assuming it doesn't but the documentation to clarify this.

Reply:
You're right, this should be doc'd. For the record, it doesn't and because we're last write wins now it could theoretically end up overwriting any existing key that's there in the store when saved if there's any kind of concurrency going on.

Reply:
I commented it that way.

---
In reply to: [179846161](https://github.com/Microsoft/botbuilder-dotnet/pull/376#discussion_r179846161) [](ancestors = 179846161)